### PR TITLE
Bump CustomBuildTasks to .NET 4.0

### PR DESCRIPTION
**Pull request by carlosmn**
The bindings are already here, and this also allows us to build both on
mono 3 as well as mono 4, which dropped pre-4.0 assemblies from the
install.

---

This is just #1034 but this is in a branch which I can push to :)

### CODE REVIEW

**Comment by nulltoken** [comment]
@bording Thoughts?

File: Lib/CustomBuildTasks/CustomBuildTasks.csproj [hunk 1]
```diff
@@ -9,9 +9,10 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>CustomBuildTasks</RootNamespace>
     <AssemblyName>CustomBuildTasks</AssemblyName>
-    <TargetFrameworkVersion>v3.5</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
     <TargetFrameworkProfile />
+    <ProductVersion>8.0.30703</ProductVersion>
+    <SchemaVersion>2.0</SchemaVersion>
```
Comment (bording) [inline Lib/CustomBuildTasks/CustomBuildTasks.csproj]: With these two lines, and and the missing TargetFrameworkVersion property, this feels like the csproj was created with an older version of VS. 

I'd recommend dumping these two lines and adding the `<TargetFrameworkVersion>v4.0</TargetFrameworkVersion>` line instead.

**Comment by bording** [comment]
I did the exact same thing last night, but saw it was done in the xunit20 branch already, so I didn't push it up as a PR!

#1034 was done before I added `GenerateNativeDllNameTask` but this one is good since it's off of current vNext.

Other than the tweak to the csproj I mentioned above, LGTM.

**Comment by Therzok** [comment]
I think this has been iterated like 4 times now. :dango:

**Comment by nulltoken** [comment]
@carlosmn Thanks for reminding me of this!

I've rebased/updated #1034 and eventually merged it.

**Pull request closed without merging**